src/lib/KQStream.ts

```diff
@@ -72,43 +73,50 @@ type StreamEvents = {
 type Events = GameEvents & StreamEvents;
 
 export class KQStream extends ProtectedEventEmitter<Events> {
-    /**
-     * Some keys in websocket messages are not formatted correctly.
-     * This object is used by `KQStream#parse` to fix them.
-     */
-    private static keyAdjustments = {
-        'getOnSnail: ': 'getOnSnail',
-        'getOffSnail: ': 'getOffSnail'
-    };
-
     private options: KQStreamOptions;
     private connection: websocket.connection;
 
     /**
      * Parses a Killer Queen websocket message.
-     * 
+     *
      * @param message The message to parse
      * @returns The type (key) and data (value) of the message,
      *          or `undefined` if unable to parse the message.
      */
     static parse(message: string) {
         // Websocket messages use format `![k[${key}],v[${value}]]!`
         // Replace `${key}` with the key, and `${value}` with the value
-        const dataArray = message.match(/!\[k\[(.*?)\],v\[(.*)?\]\]!/);
+        const dataArray = message.match(/!\[k\[(.+)\],v\[(.*)?\]\]!/);
         if (dataArray === null) {
             return;
         }
         const parsedMessage = {
-            key: dataArray[1],
-            value: dataArray[2]
+            key: KQStream.normalizeKey(dataArray[1]),
+            value: KQStream.normalizeValues(dataArray[2]),
         };
-        const newKey = KQStream.keyAdjustments[parsedMessage.key];
-        if (newKey !== undefined) {
-            parsedMessage.key = newKey;
-        }
+
         return parsedMessage;
     }
 
+    static normalizeKey(key: string) {
+        // Replace non letters with empty
+        return key.replace(/[^a-z]/gi, "").trim();
+    }
+
+    static normalizeValues(values: any) {
+        const valuesList = values.split(",");
+        const normalizedValuesList = _.map(valuesList, (value) => {
+            // If it's a number, no processing necessary
+            if (!Number.isNaN(Number(value))) { return value; }
+
+            // Convert to camelCase
+            value = value.replace(/^Red$/, "Gold");
```

Comment:
Leave a comment about why we change "Red" to "Gold". You can repurpose the leftover comment in `parsers.ts`.